Detectors/TPC/simulation/run/convertDigitsToRawZS.cxx

```diff
@@ -106,13 +96,37 @@ void convertDigitsToZSfinal(std::string_view digitsFile, std::string_view output
       writer.registerLink(feeid, cruID, defaultLink, j & 1, fmt::format("{}cru{}.raw", outDir, cruID));
     }
   }
-  for (Long64_t ievent = 0; ievent < treeSim->GetEntries(); ++ievent) {
-    treeSim->GetEntry(ievent);
 
+  for (int iSecBySec = 0; iSecBySec < Sector::MAXSECTOR; ++iSecBySec) {
+    DigitArray inputDigits;
+    treeSim->ResetBranchAddresses();
     for (int iSec = 0; iSec < Sector::MAXSECTOR; ++iSec) {
-      inputDigits[iSec] = *vDigitsPerSectorCollection[iSec]; //????
+      if (sectorBySector) {
+        iSec = iSecBySec;
+      }
+      vDigitsPerSectorCollection[iSec] = nullptr;
+      treeSim->SetBranchAddress(TString::Format("TPCDigit_%d", iSec), &vDigitsPerSectorCollection[iSec]);
+      if (sectorBySector) {
+        break;
+      }
+    }
+    for (Long64_t ievent = 0; ievent < treeSim->GetEntries(); ++ievent) {
+      treeSim->GetEntry(ievent);
```

Comment:
Instead of reading the whole tree, could you try to read the needed branch only ?
````
if (sectorBySector) {
  treeSim->GetBranch( TString::Format("TPCDigit_%d", iSec) )->GetEntry(ievent);
}
else {
  treeSim->GetEntry(ievent);
}
````

Reply:
ok, didn't know that, makes sense